src/AutoMapper/MappingEngine.cs

```diff
@@ -212,28 +190,22 @@ public TDestination DynamicMap<TDestination>(object source)
 
         public object DynamicMap(object source, Type sourceType, Type destinationType)
         {
+            ConfigurationProvider.CreateMissingTypeMaps = true;

```

Comment:
My suggestion is instead of having CreateMissingTypeMaps be used as a bunch of if statements all throughout the code, is if you set it to true you add a profile that has a convention that's always true, so everything can map to everything.  And when you set to false it removes said convention from list.

The question I have is should this new convention not have prefix and postfixes or should it wrap around the default profile and share it's prefix and postfixes.

I can submit a PR after merge if you want.